dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/trackedentity/store/DefaultEnrollmentStore.java

```diff
@@ -80,7 +80,8 @@ public DefaultEnrollmentStore( @Qualifier( "readOnlyJdbcTemplate" ) JdbcTemplate
     {
         EnrollmentRowCallbackHandler handler = new EnrollmentRowCallbackHandler();
 
-        jdbcTemplate.query( withAclCheck( GET_ENROLLMENT_SQL_BY_TEI, ctx, " pi.programid IN (:programIds)" ),
+        jdbcTemplate.query( withIncludeDeleted(
```

Comment:
I am not sure of a better name, but when I read this code, I think

-- it will include deleted records

but it will not necessarily since its checking the `ctx.getQueryParams().isIncludeDeleted()`.

Also should that be default behavior in tracker? To only include soft-deleted records if the user wants them? If so this design of having to wrap queries with `withIncludeDeleted()` can lead to someone forgetting to do that. Not sure if there is a way to do that by default to mitigate that risk. Of course only if that is the default behavior we want.

Reply:
@stian-sandvold @Bekkalizer I am really not sure what is the default behaviour here, do you have more information?

Anyway, if the default is to exclude soft-deleted values, I can try to reverse the logic here but in general we do not have a good solution about that in the system.

Reply:
I made it explicit that includeDeleted is false.